docs/cndb_omf.md

```diff
@@ -0,0 +1,294 @@
+DRAFT!!! DRAFT!!! DRAFT!!! DRAFT!!! DRAFT!!! DRAFT!!!
+
+# OMF descriptions
+
+Describes the proposed CNDB omf for HSE-3.0.
+
+## Record types
+
+Each record type contains a header at the beginning. This header contains the type of the record and its
+size. This header is not listed in the following breakdown of the various record types.
+
+### Version
+- Magic: A magic number to verify that the MDC is a CNDB MDC.
+- Version: Version of the CNDB omf.
+- Captgt: (Capacity Target) Size of the CNDB.
+
+### Info
+There are two types of info records. `CNDB_TYPE_INFO` and `CNDB_TYPE_INFOD`.
+- Create-time parameters for a KVS
+  - Prefix length: Length of prefixes when using prefix delete.
+- Flags: Flags for the KVS
+  - `CN_CFLAG_CAPPED`: Whether or not the KVS is a capped KVS.
+- Meta size: The Info record is followed by opaque metadata. This field records the size of this metadata.
+  See the "Meta[]" field below.
+- Cnid: (CN id) A unique identifier for the KVS.
+- CN name: Name of the KVS
+- Meta[]: Opaque metadata ecoded/decoded by the caller.
+
+### CNDB meta
+- Seqno max: Max seqno of the persisted records in the KVDB.
+
+### TX
+The number of create and delete records aren't recorded here. They can be tracked as they come along. A
+txn is "complete" when the number of `TXC` records and `ACK-C` records are equal. A "complete" txn is
+one that can be rolled forward if necessary.
+
+Every transaction in CNDB has at least one `TXC` record. So the `ack-N` records do not count toward marking
+the txn as complete.
+
+- TX id: A unique id for the CNDB txn. Note that a txn can span different 'cnid's.
+- Seqno: KVDB seqno at the time of logging this record.
+- Ingest id: Every ingest operation gets a unique id. If TX does not describe an ingest operation, the
+  value is set to `CNDB_INVAL_INGESTID`.
+- Transaction horizon: Transaction horizon for an ingest. This is used by WAL to reclaim its files. When
+  the operation is not an ingest, the value is set to `CNDB_INVAL_HORIZON`.
```

Comment:
If we have the CNID in the TX record, do we need it TXC, TXM, etc?

Reply:
We do not have cnid in the TX record because for ingests, a TX spans multiple cnids. So the cnid field is part of the node/kvset level records.

We could skip cnid in TXM if it's specified in TXC. But I think I'll leave it in for now and see if I can eliminate it when implementing this.